src/App/Router.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Friendica\App;
+
+
+use FastRoute\DataGenerator\GroupCountBased;
+use FastRoute\Dispatcher;
+use FastRoute\RouteCollector;
+use FastRoute\RouteParser\Std;
+use Friendica\Module\Itemsource;
+
+class Router
+{
+	/** @var RouteCollector */
+	public $routeCollector;

```

Comment:
Please no public variable, create a `getRouteCollector()` instead.

Reply:
PhpDoc

Reply:
I'm not a fan of transparent getters that don't transform the protected/private property, what's the benefit exactly over a public property?

Reply:
The problem would be something like:

```php
public function doSomething(Router $router) 
{
    $router->routeCollection = null;
}

public function doSomethingElse(Router $router)
{
    $router->routeCollection->method();
    *BANG*
}
```

The getter doesn't allow such modification. It's part of the encapsulation principle